README.md

```diff
@@ -9,45 +9,65 @@ See [screenshots](#screenshots) below, or jump right to [setup instructions](#se
 
 ## Setup
 
-### Proof General
-
-Download and install Proof General from [GitHub](https://github.com/ProofGeneral/PG):
-
-```bash
-git clone https://github.com/ProofGeneral/PG ~/.emacs.d/lisp/PG
-cd ~/.emacs.d/lisp/PG
-make
-```
-
-Then add the following to your `.emacs`:
-
-```elisp
-;; Open .v files with Proof General's Coq mode
-(load "~/.emacs.d/lisp/PG/generic/proof-site")
-```
-
 ### MELPA
 
-MELPA is a repository of Emacs packages. Skip this step if you already use [MELPA](http://melpa.org/#/getting-started). Otherwise, add the following to your `.emacs` and restart Emacs:
+Both `proof-general` and `company-coq` are on [MELPA](https://melpa.org/#/getting-started), a repository of Emacs packages. Skip this step if you already use MELPA. Otherwise, add the following to your `.emacs`:
 
 ```elisp
 (require 'package)
-(add-to-list 'package-archives '("melpa" . "http://melpa.org/packages/") t)
+(let* ((no-ssl (and (memq system-type '(windows-nt ms-dos))
+                    (not (gnutls-available-p))))
+       (proto (if no-ssl "http" "https")))
+  (add-to-list 'package-archives
+               (cons "melpa" (concat proto "://melpa.org/packages/")) t))
 (package-initialize)
 ```
 
-### Company-Coq
+Then follow one of the following two approaches:
+
+* install Proof General and Company-Coq automatically by using
+    [use-package (see below)](#use-package);
```

Comment:
I've seen these use-package suggestions pop up here and there, but I don't use it myself, so I'd rather not commit to supporting it, or recommend it (also, pet theory: people who use us-package already know how to configure it, and to newcomers it looks intimidating)

Reply:
I've mixed feelings about that. It is true that the syntax is somewhat verbose, but AFAIAC I use it on a regular basis and I see a nice advantage in the `use-package` setup: you can install all your favorite packages immediately (e.g. if you want to install it on another PC) without requiring any `M-x ...` typing (even if `use-package` itself is not installed). You just need to add these "declarative" snippets in the `.emacs` and restart Emacs.

To sum up I think it'd be a pity not to mention it at all; but if you want maybe I could add a phrasing that the `M-x package-install ...` procedure is the *standard* one?

Actually, even if the use-package syntax is a bit verbose, I really think that it's quite easier to set up for people not necessarily familiar with both Emacs and MELPA… (I think you were saying [something similar](https://github.com/melpa/melpa/pull/5700#issuecomment-416768772) months ago :)

Reply:
I'm sure use-package is great; I just think that people who already use it don't need the extra instructions, and people who do not won't be helped much by company-coq mentioning it (since they'd have to install it the package-install way, anyway).

> (I believe you were saying something similar several months ago :)

I think explaining any install mechanism is a pain, but since use-package isn't in core it doesn't help much with that ^^

Reply:
> I just think that people who already use it don't need the extra instructions, and people who do not won't be helped much by company-coq mentioning it

OK, so I will remove that snippet from this PR to keep the README.md as simple as possible.

but I still think it'd be good that this alternative installation setup for PG+company-coq is documented at least somewhere, say in https://proofgeneral.github.io/, I guess this would be useful for some people that don't know use-package but would want to experiment a smoother installation procedure.

> (since they'd have to install it the package-install way, anyway).

Yes, but *without* running `M-x package-install use-package RET` manually, because it was bootstrapped in my snippet:

```elisp
(unless (package-installed-p 'use-package)
  (package-refresh-contents)
  (package-install 'use-package))
(eval-when-compile
  (require 'use-package))
```